File: apps/web/src/types/feeInstallmentStatus.ts
export interface FeeInstallmentStatusSummary {
  total_due: number;
  total_paid: number;
  outstanding_balance: number;
}

export type FeeInstallmentStatusValue = "Paid" | "Partial" | "Pending" | "Overdue";

export interface FeeInstallmentStatusInstallment {
  fee_installment_id: number;
  installment: string;
  category: string;
  due_date: string; // ISO date
  amount: number;
  paid: number;
  balance: number;
  status: FeeInstallmentStatusValue;
}

export interface FeeInstallmentStatusResponse {
  summary: FeeInstallmentStatusSummary;
  installments: FeeInstallmentStatusInstallment[];
}

export interface StudentSearchItem {
  id: number;
  student_name: string;
  student_code?: string | null;
  admission_no?: string | null;
  roll_no?: string | null;
  class_id?: number | null;
  class_name?: string | null;
}

File: apps/fastapi/app/schemas/fee_installment_status.py
from __future__ import annotations

from datetime import date

from pydantic import BaseModel, Field


class FeeInstallmentStatusSummary(BaseModel):
    total_due: float = Field(..., ge=0)
    total_paid: float = Field(..., ge=0)
    outstanding_balance: float = Field(..., ge=0)


class FeeInstallmentStatusItem(BaseModel):
    fee_installment_id: int
    installment: str
    category: str
    due_date: date
    amount: float = Field(..., ge=0)
    paid: float = Field(..., ge=0)
    balance: float = Field(..., ge=0)
    status: str  # Paid / Partial / Pending / Overdue


class FeeInstallmentStatusResponse(BaseModel):
    summary: FeeInstallmentStatusSummary
    installments: list[FeeInstallmentStatusItem]

